### PR TITLE
GPU device choice for D3D11 and Vulkan

### PR DESCRIPTION
**Pull request by hrydgard**
Some rather mindless coding on a train from earlier today, heh. Mostly useful for debugging the Intel implementations of these APIs when you happen to have a discrete GPU in your machine too.

Thinking about also making a fallback so you can also enter "1" or "2" to choose the first or second devices etc, if you can't guess the correct name when modifying the INI to rescue yourself from a broken driver situation, if that ever happens...

### CODE REVIEW

File: Core/Config.h [hunk 1]
```diff
@@ -153,6 +153,9 @@ struct Config {
 
 	// GFX
 	int iGPUBackend;
+	// We have separate device parameters for each backend so it doesn't get erased if you switch backends.
+	// If not set, will use the "best" device.
+	std::string VulkanDevice;
```
Comment (unknownbrackets) [inline Core/Config.h]: sVulkanDevice and sD3D11Device?  We're kinda stuck with the convention now.

-[Unknown]

Comment (hrydgard) [reply Core/Config.h]: Yea, good point.

**Review by unknownbrackets — APPROVED** [review]
One minor difference between D3D11 and Vulkan is that, it looks like D3D11 won't save the device name until you change it - but Vulkan will update the selected device name in the config.

But probably fine even if it's slightly different.

-[Unknown]

**Comment by vsub** [comment]
I was expecting that restart will be needed but ppsspp doesn't say like for the renderer.

Maybe a popup message will be nice.
Also I don't know if this is expected behaviour but in DX11 mode,I also have 3rd option..."Microsoft Basic Renderer Driver" which is actually using my intel card

**Comment by hrydgard** [comment]
Created an issue.

**Pull request merged**